src/solana/rpc/websocket_api.py

```diff
@@ -176,6 +183,52 @@ async def logs_unsubscribe(
         await self.send_data(req)
         del self.subscriptions[subscription]
 
+    async def block_subscribe(
+        self,
+        filter_: Union[RpcBlockSubscribeFilter, RpcBlockSubscribeFilterMentions] = RpcBlockSubscribeFilter.All,
+        commitment: Optional[Commitment] = None,
+        encoding: Optional[str] = None,
+        transaction_details: TransactionDetails = None,
+        show_rewards: Optional[bool] = None,
+        max_supported_transaction_version: Optional[int] = None,
+    ) -> None:
+        """Subscribe to blocks.
+
+        Args:
+            filter_: filter criteria for the blocks.
+            commitment: The commitment level to use.
+            encoding: Encoding to use.
+            transaction_details: level of transaction detail to return.
+            show_rewards: whether to populate the rewards array.
+            max_supported_transaction_version: the max transaction version to return in responses.
+        """
+        req_id = self.increment_counter_and_get_id()
+        commitment_to_use = None if commitment is None else _COMMITMENT_TO_SOLDERS[commitment]
+        encoding_to_use = None if encoding is None else _ACCOUNT_ENCODING_TO_SOLDERS[encoding]
```

Comment:
Looks like we need to use `_TX_ENCODING_TO_SOLDERS` instead:
```suggestion
        encoding_to_use = None if encoding is None else _TX_ENCODING_TO_SOLDERS[encoding]
```

https://github.com/ruiqic/solana-py/blob/4d8be0a047c7be16dea1058f2b6f9d321bea9fe5/src/solana/rpc/core.py#L98-L104